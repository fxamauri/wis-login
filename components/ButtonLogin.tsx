import React from 'react';

import * as S from './ButtonLogin.styles';

type Props = {
  title: string;
  loading?: boolean;
};

const ButtonLogin = ({
  title,
  loading = false,
}: Props & React.ButtonHTMLAttributes<HTMLButtonElement>) => {
  return <S.ButtonLogin>{loading ? <S.Loading /> : title}</S.ButtonLogin>;
};

export default ButtonLogin;
