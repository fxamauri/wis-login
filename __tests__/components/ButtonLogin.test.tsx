import { render } from '@testing-library/react';
import ButtonLogin from '../../components/ButtonLogin';
describe('ButtonLogin', () => {
  it('Should render button', () => {
    const { container } = render(<ButtonLogin title="Entrar" loading={false} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render Button in not loading mode', () => {
    const { getByText } = render(<ButtonLogin title="Entrar" loading={false} />);

    expect(getByText('Entrar').textContent).toBe('Entrar');
  });

  it('should render Button in loading mode', () => {
    const { container } = render(<ButtonLogin title="Entrar" loading={true} />);

    expect(container.firstChild?.firstChild?.nodeName).toBe('svg');
  });
});
