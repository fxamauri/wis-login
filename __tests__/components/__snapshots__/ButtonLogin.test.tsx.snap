// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonLogin Should render button 1`] = `
<button
  class="ButtonLoginstyles__ButtonLogin-d0aenn-0 kJBLfk"
>
  Entrar
</button>
`;
